partitura/musicanalysis/note_features.py

```diff
@@ -526,6 +527,70 @@ def grace_feature(na, part, **kwargs):
     return W, feature_names
 
 
+def clef_feature(na, part, **kwargs):
+    """Clef feature
+
+    This feature encodes the current clef of the staff of each note.
+    Note that this feature does not return the staff number per note,
+    see staff_feature for this information.
+    """
+    notes = part.notes_tied if not np.all(na["pitch"] == 0) else part.rests
+    numerical_clef_dict ={
+        'G':0, 'F':1, 'C':2, 
+        'percussion':3, 'TAB':4, 
+        'jianpu':5,  'none':6}
+    names = [
+        "clef_sign",
+        "clef_line",
+        "clef_octave_change"
+    ]
+    clef_dict = defaultdict(list)
+    staff_numbers = set()
+    clef_list = [clef for clef in part.iter_all(score.Clef)]
+    if len(clef_list) > 0:
+        for clef in clef_list:
+            staff = clef.staff or 1
+            staff_numbers.add(staff)
+            time_key = "time_"+str(staff)
+            clef_key = "clef_"+str(staff)
+            clef_dict[time_key].append(clef.start.t)
+            clef_dict[clef_key].append(clef)
+
+        for staff in staff_numbers:
+            time_key = "time_"+str(staff)
+            interpolator_key = "interp_"+str(staff)
+            start_times = np.array(clef_dict[time_key])
+            clef_indices = np.arange(len(start_times))
+            interpolator = interp1d(start_times, clef_indices, 
+                                    kind = "previous", 
+                                    bounds_error=False, 
+                                    fill_value="extrapolate")
+            clef_dict[interpolator_key].append(interpolator)
+
+        W = np.zeros((len(notes), 3))
+
+        for i, n in enumerate(notes):
+            staff = n.staff or 1
```

Comment:
By iterating `notes = part.notes_tied` there is no guarantee that the order of the note/features will match the one in `note_array`. Maybe you can add a double check about this where you check if the `n.id == na[i]["id"]` and if not the same then seek the correct note to update the features so it follows the order of the `note_array`.